n64.mk

```diff
@@ -107,7 +107,8 @@ $(BUILD_DIR)/%.o: $(SOURCE_DIR)/%.S
 		DATASECTION="$(basename $@).data"; \
 		BINARY="$(basename $@).elf"; \
 		echo "    [RSP] $<"; \
-		$(N64_CC) $(RSPASFLAGS) -nostartfiles -Wl,-Ttext=0x1000 -Wl,-Tdata=0x0 -Wl,-e0x1000 -o $$BINARY $<; \
+		$(N64_CC) $(RSPASFLAGS) -nostartfiles -Wl,-Trsp.ld -Wl,--gc-sections -o $@ $<; \
+		mv "$@" $$BINARY; \
```

Comment:
It happened often enough to me to have a truncated / zero byte .o file when something went wrong. I'm not sure if it's an artifact of how GCC handles assembly files (maybe when doing .S -> .o, it opens the output file immediately), because it never happens with .o files coming from .c files.